src/tools/ceph_objectstore_tool.cc

```diff
@@ -239,12 +232,9 @@ struct pgid_object_list {
         cout << std::endl;
       }
       f->open_array_section("pgid_object");
-      string pgid = i->first.c_str();
-      std::size_t pos = pgid.find("_");
-      if (pos == string::npos)
-        f->dump_string("pgid", pgid);
-      else
-        f->dump_string("pgid", pgid.substr(0, pos));
+      spg_t pgid;
+      if (i->first.is_pg(&pgid))
+        f->dump_string("pgid", stringify(pgid));
```

Comment:
The old code would output "meta" for that collection, but I can fix it when my wip-cot-meta goes in if it does after this one.

It is annoying that if it is a pg then we want the pgid otherwise we want the name of the collection.  How about this?

if (i->first.is_pg(&pgid)) {
     f->dump_string("pgid", stringify(pgid));
else
    i->first.dump(f);   // This will dump the field as a "name" I would have preferred "coll" or something similar